### ISSUE TITLE
Support Probability Distribution Labels in PyTorchClassifier

### ISSUE THREAD

**Issue by f4str** [Trusted-AI/adversarial-robustness-toolbox#1966]
**Is your feature request related to a problem? Please describe.**
PyTorch's `nn.CrossEntropyLoss` function supports probability distribution labels. However, since the `PyTorchClassifier` always sets `_reduce_labels = True` when using this loss function, it is not possible to use probability distribution labels. The solution is to never reduce labels for this loss function as the result is the same when using vector labels or one-hot encoded labels.

**Describe the solution you'd like**
When calling `classifier.fit()` or `classifier.compute_loss()` using `nn.CrossEntropyLoss`, labels passed as a probability distribution should be kept as is and not reduced. No clear solution has been decided yet, instead ideas are below.

**Describe alternatives you've considered**
Create a new `_probability_labels` property in the `PyTorchClassifier` and have it set to True when using `nn.CrossEntropyLoss`. This will allow to check in the `reduce_labels()` function if the provided labels are one-hot encoded or probability distribution labels. However, this would introduce additional computation time to check the labels and may also break some of the tests.

Manually checking the labels beforehand to determine whether to reduce labels. But this would introduce extra complexity and everything should be left inside the `reduce_labels()` function.

Setting the `_reduce_labels` property to False in the `PyTorchClassifier` when using `nn.CrossEntropyLoss` so labels are always reduced is also an option since the results are the same. However, this breaks some of the unit tests for the PyTorch randomized smoothing implementation.

Use the `kwargs` in the `.fit()` and `.compute_loss()` functions to pass parameters to override the current `_reduce_labels` and `_int_labels` fields. This will preserve compatibility, but is not the most elegant solution.

**Additional context**
N/A
